documentation/Database-Schema-Review-And-Consolidations.md

```diff
@@ -0,0 +1,127 @@
+
+`Conform (Approve)` -> All of the applicant's responses must be marked as approved by reviewer, and consolidator agreed with reviewers decision
+
+`LOQ` -> At least one of applicant's responses marked as non conformity (Decline) and consolidator agreed with reviewers decisions. They have an option to select from LOQ or Non-Conformity
+
+`Non-conform (Decline)` -> At least one of applicant's responses marked as non conformity (Decline) and consolidator agreed with reviewers decisions. They have an option to select from LOQ or Non-Conformity
+
+`Changes Requested` -> Applies to consolidation, if at least one disagreement with reviewer below consolidator's level, this is the only option available
+
+## Review Assignment
+
+These records are created by back end when review level is reached (either through application submission or review submission, see Review Life Cycle diagram below). `Template Permission` records, alongside `permission name`, `permission name join` and `permission policy` are used to created relevant `review assignments`
+They can be queried to:
+
+- Determine who can be assigned to an application stage and level <- front end task
+- Determine if an application stage and level is fully assigned (using review_question_assignment link) <- through application list view
+- Determine if review can be started/created <- front end can check if review can be created (including self assignment) See `Review and Consolidation URL flow` diagram below
+
+#### Review Assignment Status
+
+`Available` -> can be assigned by assigner
+
+`Not Available` -> cannot be assigned by assigner, in case of `available for self assignment`, when reviewer self assigns a review, all other `review assignments` for current `stage` and `level` are changed to this status
+
+`Assigned` -> review is assigned to a user and can be started (in case of self assignment when review is self assigned, review assignment will changes status to assigned)
+
+`Available for self assignment` -> review can be started and self assigned
+
+`Assignment NOTE` -> For MVP, only level 1 reviewer can be assigned questions (vs self assignment of review), level > 1 will be self assigned (level 1 can also be configured to self assigned, in this case they will need to be configured to review all sections)
+
+##### Other fields (that are not straight away self explanatory)
+
+`assigner_id` -> null until assigned, assigner id when assigned be assigner, otherwise reviewer_id when self assigned
+
+`is_last_level` -> last level for current stage (determine if review_decision is present in UI)
+
+`allowable_section_ids` -> an array of section IDs that reviewer has permission to review (would typically have all sections), assigner can only assign questions from sections that are in this list
```

Comment:
At some point we agreed that when no sections were specified we would say the user can be assigned to any.
Are we changing to have all listed at all non-restricted `review_assignments`?